packages/ariakit/src/combobox/__examples__/combobox-cancel/test.tsx

```diff
@@ -0,0 +1,46 @@
+import { click, getByRole, press, render } from "ariakit-test-utils";
+import { axe } from "jest-axe";
+import Example from ".";
+
+const getCombobox = () => getByRole("combobox");
+const getCancelButton = () => getByRole("button", { name: "Clear input" });
+const getOption = (name: string) => getByRole("option", { name });
+const expectComboboxClearedAndFocused = () => {
+  expect(getCombobox()).toHaveValue("");
+  expect(getCombobox()).toHaveFocus();
+};
+const selectAnOption = async () => {
+  await click(getCombobox());
+  expect(getCombobox()).toHaveValue("");
+  await click(getOption("Apple"));
+  expect(getCombobox()).toHaveFocus();
+  expect(getCombobox()).toHaveValue("Apple");
+};
```

Comment:
I'd recommend against these abstractions in this case. It's okay to repeat them on each test. The problem with abstracting this way is that failures will generate ambiguous logs. The first lines will not point to the exact tests.